src/chocolatey/infrastructure.app/services/NugetService.cs

```diff
@@ -1045,12 +1045,45 @@ private void set_package_names_if_all_is_specified(ChocolateyConfiguration confi
                 config.Input = string.Empty;
 
                 var localPackages = list_run(config, logResults: false);
+                var packagesToUpdate = localPackages.Select(p => p.Key).ToList();
+
+                if (!String.IsNullOrEmpty(config.UpgradeCommand.PackageNamesToSkip))
+                {
+                    var packagesToSkip = config.UpgradeCommand.PackageNamesToSkip
+                        .Split(',')
+                        .Select(x => x.Trim())
+                        .Where(x => !String.IsNullOrEmpty(x))
+                        .ToList();
+
+                    var unknownPackagesToSkip = packagesToSkip
+                        .Where(x => !packagesToUpdate.Contains(x, StringComparer.OrdinalIgnoreCase))
+                        .ToList();
+
+                    if (unknownPackagesToSkip.Any())
+                    {
+                        this.Log().Warn(() => "Some packages specified in '-except' were not found in the local packages: '{0}'".format_with(String.Join(",", unknownPackagesToSkip)));

```

Comment:
Because except can be specified as `--except` (which is preferred), `/except`, `-except`, it may be preferable to use the name of the configuration variable `Except` or `Some packages specified to skip were not found in the local packages: ...`